tokio/src/signal/registry.rs

```diff
@@ -2,22 +2,34 @@
 
 use crate::signal::os::{OsExtraData, OsStorage};
 
-use crate::sync::mpsc::Sender;
+use crate::sync::watch::{channel, Receiver, Sender};
 
 use once_cell::sync::Lazy;
 use std::ops;
 use std::pin::Pin;
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::Mutex;
 
 pub(crate) type EventId = usize;
 
 /// State for a specific event, whether a notification is pending delivery,
 /// and what listeners are registered.
-#[derive(Default, Debug)]
+#[derive(Debug)]
 pub(crate) struct EventInfo {
     pending: AtomicBool,
-    recipients: Mutex<Vec<Sender<()>>>,
+    tx: Sender<()>,
+    rx: Receiver<()>,
+}
+
+impl Default for EventInfo {
+    fn default() -> Self {
+        let (tx, rx) = channel(());
```

Comment:
Very nit-picky, but I would reference this as `watch::channel()`, that lets the reader know the kind of channel w/o having to look at imports.